### PR TITLE
Utilize new ES fleet polling API for global checkpoint monitoring

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Utilize new ES fleet polling API for global checkpoint monitoring
https://github.com/elastic/elasticsearch/pull/71093

Tested with the agent actions and osquerybeat, all works.

The tests will fail until the new ES API is merged.
Posting is as draft until then.

## Why is it important?

This is more efficient that polling every second as we do right now.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by aleksmaus** [comment]
Updated implementation to the latest API changes, tested end-to end with agent/osquerybeat.

**Comment by aleksmaus** [comment]
The fleet system index polling API is available now with the latest elasticsearch docker image, just verified the integration tests are passing now. Opening up for code review.

**Review by blakerouse — APPROVED** [review]
Looks good.

Didn't find anything that stood out. I want to do some testing with service tokens, once this is in to ensure those work with this as well. But no need to block on that.

File: internal/pkg/monitor/monitor.go [hunk 1]
```diff
@@ -247,53 +273,29 @@ func (m *simpleMonitorT) notify(ctx context.Context, hits []es.HitT) int {
 		select {
 		case m.outCh <- hits:
 			maxVal := hits[sz-1].SeqNo
-			m.storeCheckpoint(maxVal)
+			m.storeCheckpoint([]int64{maxVal})
```
Comment (scunningham) [inline internal/pkg/monitor/monitor.go:276]: type seqNo?

**Review by scunningham — APPROVED** [review]
Looks ok.  I'm not crazy about the special configuration casing on es.NewClient.  I think we need to rethink how that works.

**Comment by aleksmaus** [comment]
> Looks ok. I'm not crazy about the special configuration casing on es.NewClient. I think we need to rethink how that works.

I'll add to my todo list past FF: to refactor the new client initialization

**Pull request merged**